From 7044f33e82425f8b839ccccc1cc32590ba82677d Mon Sep 17 00:00:00 2001
From: Shadi Ammouri <shadi@marvell.com>
Date: Mon, 16 Oct 2017 09:59:57 +0300
Subject: [PATCH 2032/2241] dma: mv_xor_v2: Revert usage of musdk-mode in xor
 driver

This reverts commit 052ade0cab26 ("dma: mv_xor_v2: Add DT property to
skip DMA registeration for user-space driver usage")
Using musdk-mode property is the wrong way to go, the proper solution
is to unbind the device at runtime, and re-bind it to the proper driver.

Change-Id: I489d7c221521903bc0eb12eff7daaa85b877c5ba
Signed-off-by: Shadi Ammouri <shadi@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/45153
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 .../devicetree/bindings/dma/mv-xor-v2.txt          | 26 +---------------------
 drivers/dma/mv_xor_v2.c                            |  5 -----
 2 files changed, 1 insertion(+), 30 deletions(-)

diff --git a/Documentation/devicetree/bindings/dma/mv-xor-v2.txt b/Documentation/devicetree/bindings/dma/mv-xor-v2.txt
index 772e425..006699e 100644
--- a/Documentation/devicetree/bindings/dma/mv-xor-v2.txt
+++ b/Documentation/devicetree/bindings/dma/mv-xor-v2.txt
@@ -8,19 +8,9 @@ Required properties:
 - dma-bus-width: DMA address bus width
 - msi-parent: Phandle to the MSI-capable interrupt controller used for
   interrupts.
-- clocks: One or more clock gating clocks to be used to enable DMA engine
+- clocks: One ore more clock gating clocks to be used to enable DMA engine
   unit in the SoC.
 
-Optional properties:
-The below is relevant only if Marvell Userspace SDK (MUSDK) is in use:
-- musdk-mode (bool): Indicates whether the device is to be controlled by musdk framework.
-  In this case, the probe process of the device is skipped, in order not to create
-  a contention with the user-space drivers.
-  In some cases, when musdk mode is enabled, the device will be attached to a generic
-  uio instance, and this is why its not marked as "status=disabled".
-  For more info about attaching the device to generic-uio instance, see
-  Documentation/devicetree/bindings/uio/uio_pdrv_genirq.txt
-
 Example:
 
 	xor0@400000 {
@@ -32,17 +22,3 @@ Example:
 		dma-coherent;
 		dma-width = <40>;
 	};
-
-
-	/* musdk-mode enabled example */
-	xor0@400000 {
-		compatible = "marvell,mv-xor-v2";
-		reg = <0x400000 0x1000>,
-		      <0x410000 0x1000>;
-		msi-parent = <&gic_odmi>;
-		clocks = <&gateclk 14>;
-		dma-coherent;
-		musdk-mode;
-		uio-name = "uio_dma_xor_0";
-		dma-width = <40>;
-	};
diff --git a/drivers/dma/mv_xor_v2.c b/drivers/dma/mv_xor_v2.c
index e6fcaff..965ad0a 100644
--- a/drivers/dma/mv_xor_v2.c
+++ b/drivers/dma/mv_xor_v2.c
@@ -890,11 +890,6 @@ static int mv_xor_v2_probe(struct platform_device *pdev)
 	struct msi_desc *msi_desc;
 	u32 dma_bus_width;
 
-	if (of_property_read_bool(pdev->dev.of_node, "musdk-mode")) {
-		dev_notice(&pdev->dev, "Marvell Version 2 XOR driver - MUSDK mode\n");
-		return -1;
-	}
-
 	dev_notice(&pdev->dev, "Marvell Version 2 XOR driver\n");
 
 	xor_dev = devm_kzalloc(&pdev->dev, sizeof(*xor_dev), GFP_KERNEL);
-- 
2.7.4

